tests/test_incremental.py

```diff
@@ -60,9 +68,15 @@ def test_incremental_basic(scheduler):
         result = clf.predict(X)
         expected = est2.predict(X)
         assert isinstance(result, da.Array)
-        rel_error = np.linalg.norm(result - expected)
-        rel_error /= np.linalg.norm(expected)
-        assert rel_error < 0.2
+        if not dataframes:
+            rel_error = np.linalg.norm(result - expected)
+            rel_error /= np.linalg.norm(expected)
+            assert rel_error < 0.2
+        else:
+            with pytest.raises(
+                ValueError, match="operands could not be broadcast together"
+            ):
+                np.linalg.norm(result - expected)
```

Comment:
What's the type of `result` here? A dask array with unknown chunks or a dask dataframe?

if it's a dataframe, I'd prefer to convert it to an array and make the same assertions, so that we aren't vulnerable to changes in pandas' broadcasting rules.

Reply:
Thanks for that. That cleans up the tests a lot.